### PR TITLE
Mark mappings along with root viewmodels - fixes #2574

### LINKED ISSUE #2574: Calling update on component does not update mapping
When calling update on a component where the data has changed on a nested property, that changed data is not updated and thus the view does not update.

Affects 0.7.3

See jsFiddle to reproduce: [http://jsfiddle.net/jpas/xcumonru/15/](url)
Click run, then click redraw row and look at the dev console. You will see that both key-paths (which should point to the same object) in fact point to different objects.

Thanks.

### PR DESCRIPTION
**Pull request by evs-chris**
**Description of the pull request:**
This adds a step to root model mark that iterates the mappings and marks them as well.

**Fixes the following issues:**
#2574

**Is breaking:**
Don't think so.

**Reviewers:**
Please.

### CODE REVIEW

**Comment by martypdx** [comment]
Does that mark call happen on any upstream change from actual children? My concern would be that every change in the component data now causes cascading changes up to all mapped data.

Perhaps it should only be applied in the case of `ractive.update()` with no keypath specified.

**Comment by evs-chris** [comment]
`mark` only goes downstream at present. Would it be safer to go ahead and just move this to the update method in case that changes in the future?

**Comment by martypdx** [comment]
Ah yes, it goes upstream [here](https://github.com/ractivejs/ractive/blob/dev/src/model/Model.js#L419-L427). I think it's fine, just wanted to check that we weren't spamming the mappings hierarchy. :)

File: src/model/RootModel.js [hunk 1]
```diff
@@ -131,6 +131,17 @@ export default class RootModel extends Model {
 		return remapped;
 	}
 
+	mark () {
+		const keys = Object.keys( this.mappings );
+
+		let i = keys.length;
+		while ( i-- ) {
+			this.mappings[ keys[i] ].mark();
+		}
```
Comment (fskreuz) [inline src/model/RootModel.js]: If order doesn't matter
```js
Object.keys( this.mappings ).forEach( key => this.mappings[ key ].mark() );
```

Comment (evs-chris) [inline src/model/RootModel.js]: Fair enough :smile:. For some reason I default to performance mode, which is definitely premature optimization here.

**Pull request merged**